app/scripts/modules/netflix/pipeline/stage/isolatedTestingTarget/isolatedTestingTargetStage.ts

```diff
@@ -0,0 +1,252 @@
+            oldSecureVip = oldSecureVip || health.secureVipAddress;
+          });
+
+          const newVip = this.generateNewVip(oldVip, command.freeFormDetails);
+          const newSecureVip = this.generateNewVip(oldSecureVip, command.freeFormDetails);
+          this.stage.vipOverrides[this.getClusterId(cluster)] = { oldVip, oldSecureVip, newVip, newSecureVip };
+        }
+      });
+    }
+  }
+
+  public addCluster(): void {
+    this.stage.clusters = this.stage.clusters || [];
+    this.stage.vipOverrides = this.stage.vipOverrides || {};
+    this.providerSelectionService.selectProvider(this.$scope.application).then((selectedProvider: string) => {
+      let config = this.cloudProviderRegistry.getValue(selectedProvider, 'serverGroup');
+      this.$uibModal.open({
+        templateUrl: config.cloneServerGroupTemplateUrl,
+        controller: `${config.cloneServerGroupController} as ctrl`,
+        size: 'lg',
+        resolve: {
+          title: () => 'Add Isolated Testing Target Cluster',
+          application: () => this.$scope.application,
+          serverGroupCommand: () => this.buildServerGroupCommand(selectedProvider),
+        }
+      }).result.then((command: any) => this.applyCommandToStage(command));
+    });
+  }
+
+  public editCluster(cluster: any, index: number): void {
+    cluster.provider = cluster.provider || 'aws';
+    let config = this.cloudProviderRegistry.getValue(cluster.provider, 'serverGroup');
+    this.$uibModal.open({
+      templateUrl: config.cloneServerGroupTemplateUrl,
+      controller: `${config.cloneServerGroupController} as ctrl`,
+      size: 'lg',
+      resolve: {
+        title: () => 'Configure Isolated Testing Target Cluster',
+        application: () => this.$scope.application,
+        serverGroupCommand: () => {
+          return this.serverGroupCommandBuilder.buildServerGroupCommandFromPipeline(this.$scope.application, cluster)
+            .then((command: any) => this.configureServerGroupCommandForEditing(command));
+        }
+      }
+    }).result.then((command: any) => {
+        this.stage.clusters[index] = this.awsServerGroupTransformer.convertServerGroupCommandToDeployConfiguration(command);
+      });
+  }
+
+  public deleteCluster(index: number): void {
+    this.stage.clusters.splice(index, 1);
+  }
+
+  public editVip(cluster: any, vipType: 'oldVip' | 'newVip'): void {
+    const clusterVip = this.stage.vipOverrides[this.getClusterId(cluster)];
+    this.$uibModal.open({
+      templateUrl: require('./editVip.modal.html'),
+      controller: 'EditVipModalCtrl as vm',
+      size: 'md',
+      resolve: {
+        vip: () => clusterVip[vipType]
+      }
+    }).result.then(newVip => {
+      clusterVip[vipType] = newVip;
+    });
+  }
+
+  private getClusterId(cluster: any): string {
+    return `${this.getRegion(cluster)}::${cluster.account}::${this.namingService.getClusterName(cluster.application, cluster.stack, cluster.freeFormDetails)}`;
```

Comment:
I think you can replace `this.namingService.getClusterName(...)` with `this.getClusterName(cluster)`, which you defined above (makes it a little easier to read)